src/vshaxe/commands/Commands.hx

```diff
@@ -72,6 +73,44 @@ class Commands {
 		});
 	}
 
+	function extendedTyping(args) {
+		if (args.text == "{")
+			indentCurlyBracket();
+		commands.executeCommand('default:type', args);
+	}
+
+	function indentCurlyBracket():Void {
+		var editor = window.activeTextEditor;
+		if (editor == null)
+			return;
+		var lines:Array<{range:Range, spaces:String}> = [];
+		for (selection in editor.selections) {
+			if (!selection.isEmpty)
+				continue;
+			if (selection.active.line == 0)
+				continue;
+			var line = editor.document.lineAt(selection.active.line);
+			if (!line.isEmptyOrWhitespace)
+				continue;
+			var prevLine = editor.document.lineAt(selection.active.line - 1);
+			if (prevLine.text.length > 0) {
+				var char = prevLine.text.fastCodeAt(prevLine.text.length - 1);
+				if (char == "{".code || char == "[".code || char == "(".code)
```

Comment:
This is not a very safe check, what if there's some whitespace at the end of the line? Also, a comment explaining what exactly it's for might be helpful.